lib/rules/prefer-node-protocol.js

```diff
@@ -0,0 +1,151 @@
+                "enforce using the `node:` protocol when importing Node.js builtin modules.",
+            recommended: true,
+            url: "https://github.com/eslint-community/eslint-plugin-n/blob/HEAD/docs/rules/prefer-node-protocol.md",
+        },
+        fixable: "code",
+        messages: {
+            [messageId]: "Prefer `node:{{moduleName}}` over `{{moduleName}}`.",
+        },
+        schema: [
+            {
+                type: "object",
+                properties: {
+                    version: getConfiguredNodeVersion.schema,
+                },
+                additionalProperties: false,
+            },
+        ],
+        type: "suggestion",
+    },
+    create(context) {
+        function isCallExpression(node, { name, argumentsLength }) {
+            if (node?.type !== "CallExpression") {
+                return false
+            }
+
+            if (node.optional) {
+                return false
+            }
+
+            if (node.arguments.length !== argumentsLength) {
+                return false
+            }
+
+            if (
+                node.callee.type !== "Identifier" ||
+                node.callee.name !== name
+            ) {
+                return false
+            }
+
+            return true
+        }
+
+        function isStringLiteral(node) {
+            return node?.type === "Literal" && typeof node.type === "string"
+        }
+
+        function isStaticRequire(node) {
+            return (
+                isCallExpression(node, {
+                    name: "require",
+                    argumentsLength: 1,
+                }) && isStringLiteral(node.arguments[0])
+            )
+        }
+
+        function isEnablingThisRule(context, node) {
+            const version = getConfiguredNodeVersion(context)
+
+            const supportedVersionForEsm = "^12.20.0 || >= 14.13.1"
+            // Only check Node.js version because this rule is meaningless if configured Node.js version doesn't match semver range.
+            if (!version.intersects(getSemverRange(supportedVersionForEsm))) {
+                return false
+            }
+
+            const supportedVersionForCjs = "^14.18.0 || >= 16.0.0"
+            // Only check when using `require`
+            if (
+                isStaticRequire(node.parent) &&
+                node.parent.arguments[0] === node &&
+                !version.intersects(getSemverRange(supportedVersionForCjs))
+            ) {
+                return false
+            }
+
+            return true
+        }
+
+        const targets = []
+        return [
+            visitImport(context, { includeCore: true }, importTargets => {
+                targets.push(...importTargets)
+            }),
+            visitRequire(context, { includeCore: true }, requireTargets => {
+                targets.push(
+                    ...requireTargets.filter(target =>
+                        isStaticRequire(target.node.parent)
+                    )
```

Comment:
Can this check not just be `isStringLiteral`?

I think it's something like this --
```suggestion
                    ...requireTargets.filter(target =>
                        isStringLiteral(target.node.parent.arguments[0])
                    )
```

Reply:
I think `isStaticRequire` is needed here.
If `isStringLiteral` is used, some test codes (e.g. `const fs = require.resolve("fs")`) are failed.